spec/std/string_spec.cr

```diff
@@ -661,110 +661,215 @@ describe "String" do
     reversed.should eq("はちいんこ")
   end
 
-  it "gsubs char with char" do
-    replaced = "foobar".gsub('o', 'e')
-    replaced.bytesize.should eq(6)
-    replaced.should eq("feebar")
-  end
+  describe "sub" do
+    it "subs char with char" do
+      replaced = "foobar".sub('o', 'e')
+      replaced.bytesize.should eq(6)
+      replaced.should eq("feobar")
+    end
 
-  it "gsubs char with string" do
-    replaced = "foobar".gsub('o', "ex")
-    replaced.bytesize.should eq(8)
-    replaced.should eq("fexexbar")
-  end
+    it "subs char with string" do
+      replaced = "foobar".sub('o', "ex")
+      replaced.bytesize.should eq(7)
+      replaced.should eq("fexobar")
+    end
 
-  it "gsubs char with string depending on the char" do
-    replaced = "foobar".gsub do |char|
-      case char
-      when 'f'
+    it "subs char with string" do
+      replaced = "foobar".sub {|char|
+        char.should eq 'f'
         "some"
-      when 'o'
-        "thing"
-      when 'a'
-        "ex"
-      else
-        char
-      end
+      }
+      replaced.bytesize.should eq(9)
+      replaced.should eq("someoobar")
+
+      empty = ""
+      empty.sub { 'f' }.should be(empty)
```

Comment:
ah sorry didn't see it as it was embedded in another spec - would prefer for it to be a separate test case :) But of course, that are details.